hack/jenkins/e2e.sh

```diff
@@ -84,7 +82,7 @@ ${GCE_DEFAULT_SKIP_TEST_REGEX}\
 |NetworkingNew\
 |Nodes\sNetwork\
 |Nodes\sResize\
-|MaxPods\
```

Comment:
Done. Hostdir, multiport and networking haven't failed for quite some time, so I'm moving them back to ordinary suite as well. I also haven't seen failures for `identically named` services or `change the type`, but I'm reluctant to moving any of services tests back.

OTOH we're seeing failures in ordinary or parallel suite, but most of them can be contributed to Master failures fixed by #11198 or Kubelet restarts #11215.